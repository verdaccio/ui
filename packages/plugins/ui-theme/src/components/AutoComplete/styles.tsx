import styled from '@emotion/styled';
import { Theme } from 'verdaccio-ui/design-tokens/theme';

import TextField from '../TextField';

export interface InputFieldProps {
  color: string;
}

export const Wrapper = styled('div')({
  width: '100%',
  position: 'relative',
  zIndex: 1,
});

export const StyledTextField = styled(TextField)<{ theme?: Theme }>((props) => ({
  '& .MuiInputBase-root': {
    ':before': {
      content: "''",
      border: 'none',
    },
    ':after': {
      borderColor: props.theme?.palette.white,
    },
    ':hover:before': {
      content: 'none',
    },
    ':hover:after': {
      content: 'none',
      transform: 'scaleX(1)',
    },
    [`@media screen and (min-width: ${props.theme?.breakPoints.medium}px)`]: {
      ':hover:after': {
        content: "''",
      },
    },
  },
  '& .MuiInputBase-input': {
    [`@media screen and (min-width: ${props.theme?.breakPoints.medium}px)`]: {
      color: props.theme?.palette.white,
    },
  },
}));
