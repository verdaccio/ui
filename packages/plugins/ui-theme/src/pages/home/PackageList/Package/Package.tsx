import { css } from '@emotion/react';
import styled from '@emotion/styled';
import BugReport from '@mui/icons-material/BugReport';
import DownloadIcon from '@mui/icons-material/CloudDownload';
import HomeIcon from '@mui/icons-material/Home';
import React, { useCallback } from 'react';
import { useTranslation } from 'react-i18next';
import { useDispatch, useSelector } from 'react-redux';
import Grid from 'verdaccio-ui/components/Grid';
import { FileBinary, Law, Time, Version } from 'verdaccio-ui/components/Icons';
import Link from 'verdaccio-ui/components/Link';
import ListItem from 'verdaccio-ui/components/ListItem';
import Tooltip from 'verdaccio-ui/components/Tooltip';
import { Theme } from 'verdaccio-ui/design-tokens/theme';
import fileSizeSI from 'verdaccio-ui/utils/file-size';
import { formatDate, formatDateDistance, getAuthorName } from 'verdaccio-ui/utils/package';
import { isURL } from 'verdaccio-ui/utils/url';

import { Author as PackageAuthor, PackageMetaInterface } from '../../../../../types/packageMeta';
import { Dispatch, RootState } from '../../../../store/store';
import Tag from './Tag';
import {
  Author,
  Avatar,
  Description,
  Details,
  GridRightAligned,
  IconButton,
  OverviewItem,
  PackageListItemText,
  PackageTitle,
  Published,
  TagContainer,
  Text,
  Wrapper,
  WrapperLink,
} from './styles';

interface Bugs {
  url: string;
}
interface Dist {
  unpackedSize: number;
  tarball: string;
}

export interface PackageInterface {
  name: string;
  version: string;
  time?: number | string;
  author: PackageAuthor;
  description?: string;
  keywords?: string[];
  license?: PackageMetaInterface['latest']['license'];
  homepage?: string;
  bugs?: Bugs;
  dist?: Dist;
}

const Package: React.FC<PackageInterface> = ({
  author: { name: authorName, avatar: authorAvatar },
  bugs,
  description,
  dist,
  homepage,
  keywords = [],
  license,
  name: packageName,
  time,
  version,
}) => {
  const config = useSelector((state: RootState) => state.configuration.config);
  const dispatch = useDispatch<Dispatch>();
  const { t } = useTranslation();

  const handleDownload = useCallback(
    async (tarballDist: string) => {
      // FIXME: check, the dist might be different thant npmjs
      const link = tarballDist.replace(`https://registry.npmjs.org/`, config.base);
      dispatch.download.getTarball({ link });
    },
    [dispatch, config]
  );

  const renderVersionInfo = (): React.ReactNode =>
    version && (
      <OverviewItem>
        <StyledVersion />
        {t('package.version', { version })}
      </OverviewItem>
    );

  const renderAuthorInfo = (): React.ReactNode => {
    const name = getAuthorName(authorName);
    return (
      <Author>
        <Avatar alt={name} src={authorAvatar} />
        <Details>
          <Text text={name} />
        </Details>
      </Author>
    );
  };

  const renderFileSize = (): React.ReactNode =>
    dist?.unpackedSize && (
      <OverviewItem>
        <StyledFileBinary />
        {fileSizeSI(dist.unpackedSize)}
      </OverviewItem>
    );

  const renderLicenseInfo = (): React.ReactNode =>
    license && (
      <OverviewItem>
        <StyledLaw />
        {license}
      </OverviewItem>
    );

  const renderPublishedInfo = (): React.ReactNode =>
    time && (
      <OverviewItem>
        <StyledTime />
        <Published>{t('package.published-on', { time: formatDate(time) })}</Published>
        {formatDateDistance(time)}
      </OverviewItem>
    );

  const renderHomePageLink = (): React.ReactNode =>
    homepage &&
    isURL(homepage) && (
      <Link external={true} to={homepage}>
        <Tooltip aria-label={t('package.homepage')} title={t('package.visit-home-page')}>
          <IconButton aria-label={t('package.homepage')} size="large">
            <HomeIcon />
          </IconButton>
        </Tooltip>
      </Link>
    );

  const renderBugsLink = (): React.ReactNode =>
    bugs?.url &&
    isURL(bugs.url) && (
      <Link external={true} to={bugs.url}>
        <Tooltip aria-label={t('package.bugs')} title={t('package.open-an-issue')}>
          <IconButton aria-label={t('package.bugs')} size="large">
            <BugReport />
          </IconButton>
        </Tooltip>
      </Link>
    );

  const renderDownloadLink = (): React.ReactNode =>
    dist?.tarball &&
    isURL(dist.tarball) && (
      <Link
        external={true}
        onClick={() => {
          handleDownload(dist.tarball);
        }}
        to="#"
      >
        <Tooltip
          aria-label={t('package.download', { what: t('package.the-tar-file') })}
          title={t('package.tarball')}
        >
          <IconButton aria-label={t('package.download')} size="large">
            <DownloadIcon />
          </IconButton>
        </Tooltip>
      </Link>
    );

  const renderPrimaryComponent = (): React.ReactNode => {
    return (
      <Grid container={true} item={true} xs={12}>
        <Grid item={true} xs={11}>
          <WrapperLink to={`/-/web/detail/${packageName}`}>
            <PackageTitle className="package-title">{packageName}</PackageTitle>
          </WrapperLink>
        </Grid>
        <GridRightAligned
          alignItems="center"
          container={true}
          item={true}
          justify="flex-end"
          xs={true}
        >
          {renderHomePageLink()}
          {renderBugsLink()}
          {renderDownloadLink()}
        </GridRightAligned>
      </Grid>
    );
  };

  const renderSecondaryComponent = (): React.ReactNode => {
    const tags = keywords.sort().map((keyword, index) => <Tag key={index}>{keyword}</Tag>);
    return (
      <>
        <Description>{description}</Description>
        {tags.length > 0 && <TagContainer>{tags}</TagContainer>}
      </>
    );
  };

  const renderPackageListItemText = (): React.ReactNode => (
    <PackageListItemText
      className="package-link"
      primary={renderPrimaryComponent()}
      secondary={renderSecondaryComponent()}
    />
  );

  return (
    <Wrapper className={'package'} data-testid="package-item-list">
      <ListItem alignItems={'flex-start'}>{renderPackageListItemText()}</ListItem>
      <ListItem alignItems={'flex-start'}>
        {renderAuthorInfo()}
        {renderVersionInfo()}
        {renderPublishedInfo()}
        {renderFileSize()}
        {renderLicenseInfo()}
      </ListItem>
    </Wrapper>
  );
};

export default Package;

const iconStyle = ({ theme }: { theme: Theme }) => css`
  margin: 2px 10px 0 0;
  fill: ${theme?.palette.mode === 'light' ? theme?.palette.greyLight2 : theme?.palette.white};
`;

const StyledVersion = styled(Version)`
  ${iconStyle};
`;

const StyledFileBinary = styled(FileBinary)`
  ${iconStyle};
`;

const StyledLaw = styled(Law)`
  ${iconStyle};
`;

const StyledTime = styled(Time)`
  ${iconStyle};
`;
