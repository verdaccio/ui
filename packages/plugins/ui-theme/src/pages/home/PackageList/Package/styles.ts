import styled from '@emotion/styled';
import { Link } from 'react-router-dom';
import { default as Photo } from 'verdaccio-ui/components/Avatar';
import Grid from 'verdaccio-ui/components/Grid';
import { default as MuiIconButton } from 'verdaccio-ui/components/IconButton';
import Label from 'verdaccio-ui/components/Label';
import List from 'verdaccio-ui/components/List';
import ListItemText from 'verdaccio-ui/components/ListItemText';
import { default as MuiText } from 'verdaccio-ui/components/Text';
import { Theme } from 'verdaccio-ui/design-tokens/theme';

export const OverviewItem = styled('span')<{ theme?: Theme }>(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  margin: '0 0 0 16px',
  color: theme?.palette.mode === 'light' ? theme?.palette.greyLight2 : theme?.palette.white,
  fontSize: 12,
  [`@media (max-width: ${theme?.breakPoints.medium}px)`]: {
    ':nth-of-type(3)': {
      display: 'none',
    },
  },
  [`@media (max-width: ${theme?.breakPoints.small}px)`]: {
    ':nth-of-type(4)': {
      display: 'none',
    },
  },
}));

export const Published = styled('span')<{ theme?: Theme }>(({ theme }) => ({
  color: theme?.palette.mode === 'light' ? theme?.palette.greyLight2 : theme?.palette.white,
  margin: '0 5px 0 0',
}));

export const Text = styled(Label)<{ theme?: Theme }>(({ theme }) => ({
  fontSize: '12px',
  fontWeight: theme?.fontWeight.semiBold,
  color: theme?.palette.mode === 'light' ? theme?.palette.greyLight2 : theme?.palette.white,
}));

export const Details = styled('span')({
  marginLeft: '5px',
  lineHeight: 1.5,
  display: 'flex',
  flexDirection: 'column',
});

export const Author = styled('div')({
  display: 'flex',
  alignItems: 'center',
});

export const Avatar = styled(Photo)({
  width: '20px',
  height: '20px',
});

export const WrapperLink = styled(Link)({
  textDecoration: 'none',
});

export const PackageTitle = styled('span')<{ theme?: Theme }>(({ theme }) => ({
  fontWeight: theme?.fontWeight.bold,
  fontSize: 20,
  display: 'block',
  marginBottom: 12,
  color: theme?.palette.mode == 'dark' ? theme?.palette.dodgerBlue : theme?.palette.eclipse,
  cursor: 'pointer',
  [`@media (max-width: ${theme?.breakPoints.small}px)`]: {
    fontSize: 14,
    marginBottom: 8,
  },
}));

export const GridRightAligned = styled(Grid)({
  textAlign: 'right',
});

export const Wrapper = styled(List)<{ theme?: Theme }>(({ theme }) => ({
  ':hover': {
    backgroundColor:
      theme?.palette?.type == 'dark' ? theme?.palette?.secondary.main : theme?.palette?.greyLight3,
  },
}));

export const IconButton = styled(MuiIconButton)({
  padding: 6,
  svg: {
    fontSize: 16,
  },
});

export const TagContainer = styled('span')<{ theme?: Theme }>(({ theme }) => ({
  marginTop: 8,
  marginBottom: 12,
  display: 'flex',
  flexWrap: 'wrap',
  [`@media (max-width: ${theme?.breakPoints.medium}px)`]: {
    display: 'none',
  },
}));

export const PackageListItemText = styled(ListItemText)({
  paddingRight: 0,
});

export const Description = styled(MuiText)<{ theme?: Theme }>(({ theme }) => ({
  color: theme?.palette.mode === 'light' ? theme?.palette.greyDark2 : theme?.palette.white,
  fontSize: '14px',
  paddingRight: 0,
}));
