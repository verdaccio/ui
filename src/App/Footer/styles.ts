import styled from '@emotion/styled';

import { Theme } from 'verdaccio-ui/design-tokens/theme';

export const Wrapper = styled('div')<{ theme?: Theme }>(({ theme }) => ({
  background: theme?.palette.type === 'light' ? theme?.palette.snow : theme?.palette.cyanBlue,
  borderTop: `1px solid ${theme?.palette.greyGainsboro}`,
  color: theme?.palette.type === 'dark' ? theme?.palette.white : theme?.palette.nobel01,
  fontSize: '14px',
  padding: '20px',
}));

export const Inner = styled('div')<{ theme?: Theme }>(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'flex-end',
  width: '100%',
  [`@media (min-width: ${theme?.breakPoints.medium}px)`]: {
    minWidth: 400,
    maxWidth: 800,
    margin: 'auto',
    justifyContent: 'space-between',
  },
  [`@media (min-width: ${theme?.breakPoints.large}px)`]: {
    maxWidth: 1240,
  },
}));

export const Left = styled('div')<{ theme?: Theme }>(({ theme }) => ({
  alignItems: 'center',
  display: 'none',
  [`@media (min-width: ${theme?.breakPoints.medium}px)`]: {
    display: 'flex',
  },
}));

export const Right = styled(Left)({
  display: 'flex',
});

export const Love = styled('span')<{ theme?: Theme }>(({ theme }) => ({
  color: theme?.palette.love,
  padding: '0 5px',
}));
