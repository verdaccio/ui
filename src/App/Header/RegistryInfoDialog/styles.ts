import styled from '@emotion/styled';

import DialogContent from 'verdaccio-ui/components/DialogContent';
import DialogTitle from 'verdaccio-ui/components/DialogTitle';
import { Theme } from 'verdaccio-ui/design-tokens/theme';

export const Title = styled(DialogTitle)<{ theme?: Theme }>(props => ({
  backgroundColor: props.theme && props.theme.palette.primary.main,
  color: props.theme && props.theme.palette.white,
  fontSize: props.theme && props.theme.fontSize.lg,
}));

export const Content = styled(DialogContent)<{ theme?: Theme }>(({ theme }) => ({
  padding: '0 24px',
  backgroundColor: theme?.palette.background.default,
}));
