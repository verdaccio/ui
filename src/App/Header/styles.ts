import { css } from '@emotion/core';
import styled from '@emotion/styled';

import AppBar from 'verdaccio-ui/components/AppBar';
import IconButton from 'verdaccio-ui/components/IconButton';
import Link from 'verdaccio-ui/components/Link';
import Toolbar from 'verdaccio-ui/components/Toolbar';
import { Theme } from 'verdaccio-ui/design-tokens/theme';

export const InnerNavBar = styled(Toolbar)({
  justifyContent: 'space-between',
  alignItems: 'center',
  padding: '0 15px',
});

export const Greetings = styled('span')({
  margin: '0 5px 0 0',
});

export const RightSide = styled(Toolbar)({
  display: 'flex',
  padding: 0,
});

export const LeftSide = styled(RightSide)({
  flex: 1,
});

export const MobileNavBar = styled('div')<{ theme?: Theme }>(props => ({
  alignItems: 'center',
  display: 'flex',
  borderBottom: `1px solid ${props.theme && props.theme.palette.greyLight}`,
  padding: '8px',
  position: 'relative',
}));

export const InnerMobileNavBar = styled('div')<{ theme?: Theme }>(props => ({
  borderRadius: '4px',
  backgroundColor: props.theme && props.theme.palette.greyLight,
  color: props.theme && props.theme.palette.white,
  width: '100%',
  padding: '0 5px',
  margin: '0 10px 0 0',
}));

export const IconSearchButton = styled(IconButton)({
  display: 'block',
});

export const SearchWrapper = styled('div')({
  display: 'none',
  maxWidth: '393px',
  width: '100%',
});

export const NavBar = styled(AppBar)<{ theme?: Theme }>(({ theme }) => ({
  backgroundColor: theme?.palette.type === 'light' ? theme?.palette.primary.main : theme?.palette.cyanBlue,
  color: theme?.palette.white,
  minHeight: 60,
  display: 'flex',
  justifyContent: 'center',
  [`@media (min-width: ${theme?.breakPoints.medium}px)`]: css`
    ${SearchWrapper} {
      display: flex;
    }
    ${IconSearchButton} {
      display: none;
    }
    ${MobileNavBar} {
      display: none;
    }
  `,
  [`@media (min-width: ${theme?.breakPoints.large}px)`]: css`
    ${InnerNavBar} {
      padding: 0 20px;
    }
  `,
  [`@media (min-width: ${theme?.breakPoints.xlarge}px)`]: css`
    ${InnerNavBar} {
      max-width: 1240px;
      width: 100%;
      margin: 0 auto;
    }
  `,
}));

export const StyledLink = styled(Link)<{ theme?: Theme }>(({ theme }) => ({
  color: theme?.palette.white,
}));
