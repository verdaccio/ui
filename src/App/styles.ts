import { css } from 'emotion';

import colors from '../utils/styles/colors';

export const alertError = css({
  backgroundColor: `${colors.red} !important`,
  minWidth: 'inherit !important',
});

export const alertErrorMsg = css({
  display: 'flex',
  alignItems: 'center',
});

export const alertIcon = css({
  opacity: 0.9,
  marginRight: '8px',
});
