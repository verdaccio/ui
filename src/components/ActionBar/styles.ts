import styled from 'react-emotion';
import { default as MuiFab } from '@material-ui/core/Fab';

import colors from '../../utils/styles/colors';
import ListItem from '../../muiComponents/ListItem';

export const ActionListItem = styled(ListItem)({
  paddingTop: 0,
  paddingLeft: 0,
  paddingRight: 0,
});

export const Fab = styled(MuiFab)({
  '&&': {
    backgroundColor: colors.primary,
    color: colors.white,
    marginRight: '10px',
  },
});
