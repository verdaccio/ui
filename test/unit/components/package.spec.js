/**
 * Package component
 */

import React from 'react';
import { shallow } from 'enzyme';
import Package from '../../../src/webui/components/Package/index';
import Tag from '../../../src/webui/components/Tag/index';
import { Version, WrapperLink, Field, OverviewItem } from '../../../src/webui/components/Package/styles';


/**
 * Generates one month back date from current time
 * @return {object} date object
 */
const dateOneMonthAgo = () => new Date(1544377770747)

describe('<Package /> component', () => {
  test.skip('should load the component', () => {
    const props = {
      name: 'verdaccio',
      version: '1.0.0',
      time: dateOneMonthAgo(),
      license: 'MIT',
      description: 'Private NPM repository',
      author: {
        name: 'Sam'
      },
      keywords: [
        "verdaccio"
      ]
    };

    const wrapper = shallow(
      <Package {...props} />
    );


    // integration expectations

    // check link
    expect(wrapper.find(WrapperLink).prop('to')).toEqual(`detail/${props.name}`);

    // check version
    expect(wrapper.find(Version).prop('children')).toEqual(`v${props.version}`);

    // TODO - REWRITE THE TEST
    //expect(wrapper.find(Author).dive())

    // check description
    expect(wrapper.find(Field).someWhere(n => {
      return (
        n.children().first().get(0).props.children[0].props.text === 'Description' &&
        n.children().childAt(1).containsMatchingElement(<span>{props.description}</span>)
      )
    })).toBe(true);

    // check license
    expect(wrapper.find(OverviewItem).someWhere(n => n.get(0).props.children[1] === props.license)).toBe(true);

    // check keyword
    expect(wrapper.find(Tag).prop('children')).toEqual(props.keywords[0]);

  });

  test.skip('should load the component without author', () => {
    const props = {
      name: 'verdaccio',
      version: '1.0.0',
      time: dateOneMonthAgo(),
      license: 'MIT',
      author: {
        name: 'Anonymous',
        email: '',
        avatar: ''
      },
      description: 'Private NPM repository'
    };
    const wrapper = shallow(
      <Package {...props} />
    );

    // integration expectations
    expect(wrapper.html()).toMatchSnapshot();
  });
});
