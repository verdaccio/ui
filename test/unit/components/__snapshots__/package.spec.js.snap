// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Package /> component should load the component without author 1`] = `"<a class=\\"package css-scoisj e11fsc2k15\\" href=\\"detail/verdaccio\\"><div class=\\"css-esn5nr e11fsc2k0\\"><span class=\\"css-ilb586 e11fsc2k2\\"><span class=\\"css-bxt2bt e11fsc2k1\\">verdaccio</span><span class=\\"css-17xn9wj e11fsc2k5\\">v1.0.0</span></span><span class=\\"css-1dq57rh e11fsc2k4\\"><span class=\\"css-1vtxmi9 e11fsc2k3\\"><svg class=\\"e11fsc2k6 css-19sh63r ej4jd2o0\\"><title>License</title><use xlink:href=\\"[object Object]#license\\"></use></svg>MIT</span><span class=\\"css-1vtxmi9 e11fsc2k3\\"><svg class=\\"e11fsc2k6 css-1iep4l8 ej4jd2o0\\"><title>Time</title><use xlink:href=\\"[object Object]#time\\"></use></svg><span class=\\"css-c3js4s e11fsc2k7\\">Published on 09.12.2018, 17:49:30 â€¢</span>about 6 hours ago</span></span></div><div class=\\"css-tywa7u e11fsc2k9\\"><div class=\\"css-r6baau e11fsc2k8\\"><div class=\\"e11fsc2k10 css-1w35jfk e1pneb170\\">Author</div><div class=\\"css-15496ft e11fsc2k12\\"><div class=\\"MuiAvatar-root-1 MuiAvatar-colorDefault-2 css-1to0t9u e11fsc2k13\\">A</div><span class=\\"css-1xj37ub e11fsc2k11\\"><div class=\\"e11fsc2k10 css-1xe0n7g e1pneb170\\">Anonymous</div></span></div></div><div class=\\"css-r6baau e11fsc2k8\\"><div class=\\"e11fsc2k10 css-1w35jfk e1pneb170\\">Description</div><span>Private NPM repository</span></div></div></a>"`;
